### PR TITLE
`m_trademark` OK

### PR DESCRIPTION
**Pull request by AngheloAlf**
Matches `m_trademark`

Trademark does a `bzero` for `common_data` that is smaller than the size of the struct, meaning it has a substruct, so I went ahead and created said substruct and called it `Save` (inspired by OoT and MM). This change added a lot of noise to the diff. I can revert it if preferred.

### CODE REVIEW

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+    s32 temp_v0;
+
+    mCPk_InitPak(0);
+    common_data.privateInfo = &common_data.save.saveFilePrivateInfo[0];
+
+    if (mFRm_CheckSaveData() == 0) {
+        PrivateInfo* privateInfo;
+        s32 i;
+
+        // TODO: make a substruct
```
Comment (emilybrooks) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:124]:
```suggestion
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+    f32 var_fa0;
+    f32 var_fv0;
+    s16 temp_a2;
+    s16 temp_a1;
+
+    temp_a2 = this->unk_25A6A;
+    this->unk_25A6A += this->unk_25A6C;
+    temp_a1 = this->unk_25A6A;
+
+    temp_fv1 = sinf_table(temp_a1 * (f32)(M_PI / 0x8000)) * this->unk_00214;
```
Comment (Prakxo) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:191]:
```suggestion
    temp_fv1 = sinf_table(BINANG_TO_RAD(temp_a1)) * this->unk_00214;
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 3]
```diff
@@ -0,0 +1,585 @@
...
+    sp34.y = arg2;
+    sp34.z = arg3;
+    sp1C.z = 0.0f;
+    sp1C.y = 1.0f;
+    sp1C.x = 0.0f;
+    sp28.z = 0.0f;
+    sp28.y = 0.0f;
+    sp28.x = 0.0f;
+    setPerspectiveView(&this->view, 15.0f, 10.0f, 12800.0f);
+    setLookAtView(&this->view, &sp34, &sp28, &sp1C);
```
Comment (Prakxo) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:176]:
```suggestion
    setLookAtView(&this->view, &eye, &at, &up);
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+typedef enum TrademarkState {
+    /* 0 */ TRADEMARK_STATE_0,
+    /* 1 */ TRADEMARK_STATE_1,
+    /* 2 */ TRADEMARK_STATE_2,
+    /* 3 */ TRADEMARK_STATE_3,
+    /* 4 */ TRADEMARK_STATE_4,
+    /* 5 */ TRADEMARK_STATE_5,
+} TrademarkState;
+
+DoorData D_80805B20_jp = { 0x21, 4, 0, 0, { 0x884, 0xC8, 0x338 }, 0, 3, { 0, 0, 0 } };
```
Comment (Prakxo) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:38]:
```suggestion
DoorData demo_1_door_data = { SCENE_TITLE_DEMO, 4, 0, 0, { 0x884, 0xC8, 0x338 }, 0, 3, { 0, 0, 0 } };
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 5]
```diff
@@ -0,0 +1,585 @@
...
+DoorData D_80805B34_jp = { 0x21, 4, 0, 0, { 0xC92, 0x28, 0xC02 }, 0, 3, { 0, 0, 0 } };
```
Comment (Prakxo) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:39]:
```suggestion
DoorData demo_2_door_data = { SCENE_TITLE_DEMO, 4, 0, 0, { 0xC92, 0x28, 0xC02 }, 0, 3, { 0, 0, 0 } };
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 6]
```diff
@@ -0,0 +1,585 @@
...
+DoorData D_80805B48_jp = { 0x21, 4, 0, 0, { 0x845, 0xA0, 0x5D0 }, 0, 3, { 0, 0, 0 } };
```
Comment (Prakxo) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:40]:
```suggestion
DoorData demo_3_door_data = { SCENE_TITLE_DEMO, 4, 0, 0, { 0x845, 0xA0, 0x5D0 }, 0, 3, { 0, 0, 0 } };
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 7]
```diff
@@ -0,0 +1,585 @@
...
+DoorData D_80805B5C_jp = { 0x21, 4, 0, 0, { 0xB53, 0xA0, 0x44D }, 0, 3, { 0, 0, 0 } };
```
Comment (Prakxo) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:41]:
```suggestion
DoorData demo_4_door_data = { SCENE_TITLE_DEMO, 4, 0, 0, { 0xB53, 0xA0, 0x44D }, 0, 3, { 0, 0, 0 } };
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c [hunk 8]
```diff
@@ -0,0 +1,585 @@
...
+DoorData D_80805B70_jp = { 0x21, 4, 0, 0, { 0x62A, 0x28, 0x9A8 }, 0, 3, { 0, 0, 0 } };
```
Comment (Prakxo) [inline src/overlays/gamestates/ovl_trademark/m_trademark.c:42]:
```suggestion
DoorData demo_5_door_data = { SCENE_TITLE_DEMO, 4, 0, 0, { 0x62A, 0x28, 0x9A8 }, 0, 3, { 0, 0, 0 } };
```

File: src/overlays/gamestates/ovl_trademark/m_trademark.c (hunk 4 above)
Comment (AngheloAlf) [reply src/overlays/gamestates/ovl_trademark/m_trademark.c:38]: For whatever reason I thought we didn't had the original names of those symbols. Nice

**Review by emilybrooks — APPROVED** [review]

**Review by Prakxo — APPROVED** [review]

**Pull request merged**